### PR TITLE
Loadout Backpacks II

### PR DESCRIPTION
**Pull request by 1Sparky1**
The second attempt at this, after the first one had some conflict errors I couldn't resolve. The same principle, you can now get a custom bag (Backpack/Satchel/Duffel/Messenger), complete with name, description and, in the case of the generic ones, colour.

Also fixed it so that now if you have a bag in your loadout, you don't start with the one from your equipment preferences

Changelog:
- rscadd: "Added a Bags section to loadout preferences to cater to all your custom item needs. 
 - tweak: "Starting with a custom bag means you won't start with the one from your equipment preferences."

### CODE REVIEW

File: code/controllers/subsystems/job.dm [hunk 1]
```diff
@@ -752,6 +752,12 @@
 					if (leftovers)
 						leftovers += thing
 					Debug("EC/([H]): [thing] failed mask/suit/head check; leftovers=[!!leftovers]")
+				else if (G.slot == slot_back)
+					qdel(H.get_equipped_item(slot_back)) // Deletes starting backpack if you have a custom one
+					H.equip_to_slot_or_del(CI, G.slot)
+					to_chat(H, SPAN_NOTICE("Equipping you with [thing]!"))
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/job.dm:758]: I don't think this should be here

**Review by NonQueueingMatt — COMMENTED** [review]
I'm not really a fan of there being two ways to get a backpack now, this stuff should either be incorporated into the existing system or said system should be taken out.

**Comment by Sparkyhotdog** [comment]
> I'm not really a fan of there being two ways to get a backpack now, this stuff should either be incorporated into the existing system or said system should be taken out.

In terms of removing the old system, the issue would be that unless you made the loadout bags free (which I suppose is an option) people would be prompted to spend loadout points on a bag, even if they just wanted their default grey or department variant.

File: code/controllers/subsystems/job.dm (hunk 1 above)
Comment (Sparkyhotdog) [reply code/controllers/subsystems/job.dm:758]: This chuck, from the if to the chat, does exactly the same thing as with any other loadout item, except it removes the default grey or department one before. If you could explain further?

**Pull request closed without merging**